packages/widget-v2/src/pages/SwapPage/ConnectedWalletContent.tsx

```diff
@@ -28,8 +28,10 @@ export const ConnectedWalletContent = () => {
   const connectedWalletModal = useModal(ConnectedWalletModal);
 
   const formattedBalance = useMemo(() => {
+    const symbol = sourceDetails?.symbol ? sourceDetails?.symbol : "";
     if (sourceBalance?.error?.message) return "--";
-    if (sourceBalance === undefined) return `0 ${sourceDetails?.symbol}`;
+    if (sourceBalance === undefined) return;
+    if (sourceBalance === null) return `0 ${symbol}`;
```

Comment:
null and undefined are different, null means 0 balance, undefined means balance is not fetched yet

Reply:
this is because when we don't pass in any denoms, the server will not return back the denoms for the balances that are 0